src/01-dinosaur-facts.js

```diff
@@ -44,7 +56,13 @@ function getLongestDinosaur(dinosaurs) {}
  *  getDinosaurDescription(dinosaurs, "incorrect-id");
  *  //> "A dinosaur with an ID of 'incorrect-id' cannot be found."
  */
-function getDinosaurDescription(dinosaurs, id) {}
+ function getDinosaurDescription(dinosaurs, id) {
+  for (let i = 0 ; i < dinosaurs.length;i++){
+    if (id === dinosaurs[i].dinosaurId){
+      return `${dinosaurs[i].name} (${dinosaurs[i].pronunciation})\n${dinosaurs[i].info} It lived in the ${dinosaurs[i].period} period, over ${dinosaurs[i].mya[dinosaurs[i].mya.length-1]} million years ago.`
+    }
+  }return `A dinosaur with an ID of '${id}' cannot be found.`
+}
```

Comment:
You have to put together a lot of information in a very specific way to get that string right. Well done!